orc/orcarm.c

```diff
@@ -417,34 +417,14 @@ orc_arm_emit_load_reg (OrcCompiler *compiler, int dest, int src1, int offset)
 {
   orc_uint32 code;
 
-  if (compiler->is_64bit) {
-    /** LDR (immediate): unsigned offset variant using 64 bit registers
-     *    3                   2                   1
-     *  1 0 9 8 7 6 5 4 3 2 1 0 9 8 7 6 5 4 3 2 1 0 9 8 7 6 5 4 3 2 1 0
-     * +---------------------------------------------------------------+
-     * |1 1|1 1 1|0|0 1|0 1|         imm12         |    Rn   |    Rt   |
-     * +---------------------------------------------------------------+
-     */
-    code = 0xf9400000;
-    code |= ((offset/8)&0xfff) << 10; /** imm12 == offset/8 */
-    code |= (src1&0x1f) << 5;
-    code |= (dest&0x1f);
-
-    /** @todo change this function to support both register bits */
-
-    ORC_ASM_CODE(compiler,"  ldr %s, [%s, #%d]\n",
-        orc_arm64_reg_name (dest,ORC_ARM64_REG_64),
-        orc_arm64_reg_name (src1,ORC_ARM64_REG_64), offset);
-  } else {
-    code = 0xe5900000;
-    code |= (src1&0xf) << 16;
-    code |= (dest&0xf) << 12;
-    code |= offset&0xfff;
-
-    ORC_ASM_CODE(compiler,"  ldr %s, [%s, #%d]\n",
-        orc_arm_reg_name (dest),
-        orc_arm_reg_name (src1), offset);
-  }
+  code = 0xe5900000;
+  code |= (src1&0xf) << 16;
+  code |= (dest&0xf) << 12;
+  code |= offset&0xfff;
+
+  ORC_ASM_CODE(compiler,"  ldr %s, [%s, #%d]\n",
+      orc_arm_reg_name (dest),
+      orc_arm_reg_name (src1), offset);
```

Comment:
Are you reverting 64bit code of load reg?

Reply:
No, I just want to move this impl. to more generic one, `orc_arm64_emit_mem` because load/store instructions have many variants.